tools/gmhazard_scripts/db_creation/empirical_db/new_calc_emp_ds.py

```diff
@@ -0,0 +1,248 @@
+        distance_df, left_on="fault_name", right_on="fault_name"
+    )
+    max_dist = np.minimum(
+        np.interp(rupture_df.mag, MAG, DIST),
+        common.get_max_dist_zfac_scaled(site_data),
+    )
+    rupture_df = rupture_df[rupture_df["rjb"] < max_dist]
+
+    # Adding missing columns
+    # Site Parameters
+    rupture_df["vs30"] = site_data.vs30
+    rupture_df["vs30measured"] = site_data.get("vs30measured", default=False)
+    rupture_df["z1pt0"] = site_data.z1p0
+    rupture_df["z2pt5"] = site_data.z2p5
+
+    # Rupture Parameter
+    # hypo_depth is not dbot but can be used as a proxy for point source
+    # if there is only a single point then the dtop, dbot and hypo_depth
+    # are at the same point
+    rupture_df[["hypo_depth", "ztor"]] = rupture_df[["dbot", "dtop"]]
+
+    # Distance Parameter - OQ uses ry0 term
+    rupture_df[["rx", "ry0"]] = rupture_df[["rx", "ry"]].fillna(0)
+
+    return rupture_df.loc[rupture_df["tect_type"] == tect_type.name]
+
+
+def calculate_emp_ds(
+    background_sources_ffp: str,
+    site_source_db_ffp: str,
+    vs30_ffp: str,
+    output_dir: str,
+    z_ffp: Optional[str],
+    ims: Optional[List[gc.im.IMType]],
+    psa_periods: Optional[List[float]],
+    model_dict_ffp: Optional[str],
+    suffix: Optional[str] = None,
+):
+    nhm_data = gc.utils.ds_nhm_to_rup_df(background_sources_ffp)
+    rupture_df = pd.DataFrame(nhm_data["rupture_name"])
+    model_dict = empirical_factory.read_model_dict(model_dict_ffp)
+
+    with gc.dbs.SiteSourceDB(site_source_db_ffp) as distance_store:
+        # TODO: simplify the get_work() - removing MPI
+        fault_df, _, site_df, _ = common.get_work(
+            distance_store, vs30_ffp, z_ffp, None, None
+        )
+
+        # Check to see if any site's Z1.0 or Z2.5 is NaN
+        if np.any(np.isnan(site_df.z1p0.values)):
+            raise ValueError("Z1.0 cannot be NaN")
+        if np.any(np.isnan(site_df.z2p5.values)):
+            raise ValueError("Z2.5 cannot be NaN")
+
+        # Filter the site/station that the distance_store actually has
+        sites = [
+            site for site in site_df.index if distance_store.has_station_data(site)
+        ]
+        tect_types = nhm_data["tect_type"].unique()
+
+        for im_idx, im in enumerate(ims):
+            print(f"Processing IM: {im}, {im_idx + 1} / {len(ims)}")
+            for tect_idx, tect_type in enumerate(tect_types):
+                print(
+                    f"Processing Tectonic Type: {tect_type}, {tect_idx + 1} / {len(tect_types)}"
+                )
+                GMMs = empirical_factory.determine_all_gmm(
+                    classdef.Fault(tect_type=classdef.TectType[tect_type]),
+                    str(im),
+                    model_dict,
+                )
+                for GMM_idx, (GMM, _) in enumerate(GMMs):
+                    # TODO: Check those models
+                    if GMM.name in ("K_20", "K_20_NZ", "ZA_06", "CB_14", "ASK_14"):
```

Comment:
I guess we can merge it in the interim, but really this "story" isn't complete till we have interpolation and all GMMs happy.

Reply:
Maybe worth to add a print statement that those models are being skipped?